SimG4CMS/Calo/src/HGCSD.cc

```diff
@@ -148,43 +148,37 @@ uint32_t HGCSD::setDetUnitId(const G4Step * aStep) {
   ForwardSubdetector subdet = myFwdSubdet_;
 
   int layer(0), module(0), cell(0);
-  if (m_mode == HGCalGeometryMode::Square) {
+  if (touch->GetHistoryDepth() == levelT_) {
     layer  = touch->GetReplicaNumber(0);
-    module = touch->GetReplicaNumber(1);
-  } else {
-    if (touch->GetHistoryDepth() == levelT_) {
-      layer  = touch->GetReplicaNumber(0);
-      module = -1;
-      cell   = -1;
+    module = -1;
+    cell   = -1;
 #ifdef EDM_ML_DEBUG
-      edm::LogInfo("HGCSim") << "Depths: " << touch->GetHistoryDepth() 
-			     << " name " << touch->GetVolume(0)->GetName() 
-			     << " layer:module:cell " << layer << ":" 
-			     << module << ":" << cell << std::endl;
+    edm::LogInfo("HGCSim") << "Depths: " << touch->GetHistoryDepth() 
+			   << " name " << touch->GetVolume(0)->GetName() 
+			   << " layer:module:cell " << layer << ":" 
+			   << module << ":" << cell << std::endl;
 #endif
-    } else {
-      layer  = touch->GetReplicaNumber(2);
-      module = touch->GetReplicaNumber(1);
-      cell   = touch->GetReplicaNumber(0);
-    }
+  } else {
+    layer  = touch->GetReplicaNumber(2);
+    module = touch->GetReplicaNumber(1);
+    cell   = touch->GetReplicaNumber(0);
+  }
 #ifdef EDM_ML_DEBUG
-    edm::LogInfo("HGCSim") << "Depths: " << touch->GetHistoryDepth() <<" name "
-			   << touch->GetVolume(0)->GetName() 
-			   << ":" << touch->GetReplicaNumber(0) << "   "
-			   << touch->GetVolume(1)->GetName() 
-			   << ":" << touch->GetReplicaNumber(1) << "   "
-			   << touch->GetVolume(2)->GetName() 
-			   << ":" << touch->GetReplicaNumber(2) << "   "
-			   << " layer:module:cell " << layer << ":" << module 
-			   << ":" << cell <<" Material " << mat->GetName()<<":"
-			   << aStep->GetPreStepPoint()->GetMaterial()->GetRadlen()
-			   << std::endl;
+  edm::LogInfo("HGCSim") << "Depths: " << touch->GetHistoryDepth() <<" name "
+			 << touch->GetVolume(0)->GetName() 
+			 << ":" << touch->GetReplicaNumber(0) << "   "
+			 << touch->GetVolume(1)->GetName() 
+			 << ":" << touch->GetReplicaNumber(1) << "   "
+			 << touch->GetVolume(2)->GetName() 
+			 << ":" << touch->GetReplicaNumber(2) << "   "
+			 << " layer:module:cell " << layer << ":" << module 
+			 << ":" << cell <<" Material " << mat->GetName()<<":"
+			 << aStep->GetPreStepPoint()->GetMaterial()->GetRadlen();
 #endif
-    if (aStep->GetPreStepPoint()->GetMaterial()->GetRadlen() > 100000.) return 0;
-  }
-
+  if (aStep->GetPreStepPoint()->GetMaterial()->GetRadlen() > 100000.) return 0;
```

Comment:
@bsunanda , why radiation length is checked here? What means very large value? I would guess it never happens.

Reply:
This was done while doing the testbed studies. I don't remember the reason for this fix